generated/nidcpower/nidcpower/session.py

```diff
@@ -4801,11 +4801,39 @@ def fetch_multiple(self, count, timeout=hightime.timedelta(seconds=1.0)):
 
         '''
         import collections
-        Measurement = collections.namedtuple('Measurement', ['voltage', 'current', 'in_compliance'])
+        Measurement = collections.namedtuple('Measurement', ['voltage', 'current', 'in_compliance', 'channel'])
 
         voltage_measurements, current_measurements, in_compliance = self._fetch_multiple(timeout, count)
 
-        return [Measurement(voltage=voltage_measurements[i], current=current_measurements[i], in_compliance=in_compliance[i]) for i in range(count)]
+        if self._repeated_capability == '':
+            channel_names = self._get_channel_names(range(self.channel_count))
+        else:
+            # Check if the session was opened with independent_channels set to True by checking if
+            #  _get_channel_name() returns channel names with prefix
+            first_channel_name = self._get_channel_name(1)
+            if '/' in first_channel_name:
+                # If there is any repeated capabilities without prefix, the session must have only
+                # one instrument, so just get the prefix from the first channel and add to all of
+                # the repeated capabilities that are without prefix
```

Comment:
> If there is any repeated capabilities without prefix, the session must have only one instrument

This isn't always true. A user can open a session to `dev1, dev2/1-3`.

Reply:
This logic is parsing `self._repeated_capability`, which is only populated through `session.channels`, instead of `resource_name`. It should contain only channel names or ranges (either with or without prefix. Calling `session.channels['dev1, dev2/1-3']` would result in "Error -1074135008: IVI: Unknown channel or repeated capability name".